mol/molecule.py

```diff
@@ -604,6 +605,214 @@ def unit_cell_volume(self, unit='angstrom'):
+                4         C     -1.193860      0.689276      0.000000
+                5         C     -1.193860     -0.689276      0.000000
+                6         C     -0.000000     -1.378551      0.000000
+                7         H      2.132911     -1.231437     -0.000000
+                8         H      2.132911      1.231437     -0.000000
+                9         H      0.000000      2.462874     -0.000000
+               10         H     -2.132911      1.231437     -0.000000
+               11         H     -2.132911     -1.231437     -0.000000
+               12         H     -0.000000     -2.462874     -0.000000
+              Bonds:
+               (3)--1.5--(4)
+               (5)--1.5--(6)
+               (1)--1.5--(6)
+               (2)--1.5--(3)
+               (4)--1.5--(5)
+               (1)--1.5--(2)
+               (3)--1.0--(9)
+               (6)--1.0--(12)
+               (5)--1.0--(11)
+               (4)--1.0--(10)
+               (2)--1.0--(8)
+               (1)--1.0--(7)
+
+            >>> benzene.set_integer_bonds()
+            >>> print(benzene)
+              Atoms:
+                1         C      1.193860     -0.689276      0.000000
+                2         C      1.193860      0.689276      0.000000
+                3         C      0.000000      1.378551      0.000000
+                4         C     -1.193860      0.689276      0.000000
+                5         C     -1.193860     -0.689276      0.000000
+                6         C     -0.000000     -1.378551      0.000000
+                7         H      2.132911     -1.231437     -0.000000
+                8         H      2.132911      1.231437     -0.000000
+                9         H      0.000000      2.462874     -0.000000
+               10         H     -2.132911      1.231437     -0.000000
+               11         H     -2.132911     -1.231437     -0.000000
+               12         H     -0.000000     -2.462874     -0.000000
+              Bonds:
+               (3)--1.0--(4)
+               (5)--1.0--(6)
+               (1)--2.0--(6)
+               (2)--2.0--(3)
+               (4)--2.0--(5)
+               (1)--1.0--(2)
+               (3)--1.0--(9)
+               (6)--1.0--(12)
+               (5)--1.0--(11)
+               (4)--1.0--(10)
+               (2)--1.0--(8)
+               (1)--1.0--(7)
+
+        """
+        ceil = math.ceil
+        floor = math.floor
+        func_invert = {ceil: floor, floor: ceil}
+
+        def dfs(atom, func) -> None:
+            """Depth-first search algorithm for integer-ifying the bond orders."""
+            for b2 in atom.bonds:
+                if b2._visited:
+                    continue
+
+                b2._visited = True
+                b2.order = func(b2.order)  # func = ``math.ceil()`` or ``math.floor()``
+                del bond_dict[b2]
+
+                atom_new = b2.atom1 if b2.atom1 is not atom else b2.atom2
```

Comment:
`Bond.other_end` could be used here

Reply:
Good call!

Reply:
See https://github.com/SCM-NV/PLAMS/pull/78/commits/170e1e5bcc4431aaa18a89b5af9a2b3e38ad8fce.